### ISSUE TITLE
rake aborted!    ArgumentError: wrong number of arguments (1 for 0)

### ISSUE THREAD

**Issue by vasilemb** [jperelli/Redmine-Periodic-Task#23]
when i try to:
cd /var/www/<redminedir>; rake redmine:check_periodictasks RAILS_ENV="production"

i get the error:
<pre>
rake aborted!
ArgumentError: wrong number of arguments (1 for 0)
/usr/local/rvm/gems/ruby-2.1.0/gems/activerecord-4.2.1/lib/active_record/scoping/named.rb:24:in `all'
/var/www/.............../plugins/periodictask/lib/scheduled_tasks_checker.rb:3:in `checktasks!'
/var/www/.............../plugins/periodictask/lib/tasks/periodictask.rake:11:in `block (2 levels) in <top (required)>'
/usr/local/rvm/gems/ruby-2.1.0/bin/ruby_executable_hooks:15:in `eval'
/usr/local/rvm/gems/ruby-2.1.0/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => redmine:check_periodictasks
</pre>

**Comment by vasilemb** [comment]
rake -f /var/www/tichete.teamguard.ro/Rakefile redmine:check_periodictasks RAILS_ENV="production" --trace
** Invoke redmine:check_periodictasks (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute redmine:check_periodictasks
rake aborted!
ArgumentError: wrong number of arguments (1 for 0)
/usr/local/rvm/gems/ruby-2.1.0/gems/activerecord-4.2.1/lib/active_record/scoping/named.rb:24:in `all'
/var/www/tichete.teamguard.ro/plugins/periodictask/lib/scheduled_tasks_checker.rb:3:in `checktasks!'
/var/www/tichete.teamguard.ro/plugins/periodictask/lib/tasks/periodictask.rake:11:in `block (2 levels) in <top (required)>'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/usr/local/rvm/rubies/ruby-2.1.0/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:150:in `invoke_task'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `each'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:106:in `block in top_level'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:115:in `run_with_threads'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:100:in `top_level'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:78:in `block in run'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
/usr/local/rvm/gems/ruby-2.1.0/gems/rake-10.4.2/bin/rake:33:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.1.0/bin/rake:23:in `load'
/usr/local/rvm/gems/ruby-2.1.0/bin/rake:23:in `<main>'
/usr/local/rvm/gems/ruby-2.1.0/bin/ruby_executable_hooks:15:in `eval'
/usr/local/rvm/gems/ruby-2.1.0/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => redmine:check_periodictasks

**Comment by mape2k** [comment]
Can confirm this problem under redmine 3.1...

**Comment by jperelli** [comment]
@vasilemb @mape2k I think I found the bug, could you please try to install from the "issue-23" branch?

**Comment by mape2k** [comment]
Works on 2.6, maybe i can test 3.x later...

**Comment by jperelli** [comment]
Thanks! If you confirm that this works in 3.x, then I'll merge the fix into master.

**Comment by Mycron** [comment]
@jperelli i tested your fix on redmine 3.1 with ruby 2.2.1 and subject bug is gone for me.
I had same bug as @vasilemb, and that's what i did (i use master branch all the way):
1. i watched your commit 73b40ae2e475d6166a2a8a400e52792e995422fb and copy modified string.
2. backup file scheduled_tasks_checker.rb and rewrite in it string from commit.
3. After that i ran manually task with: RAILS_ENV="production" rake --trace redmine:check_periodictasks
That is my results:
````
root@chinook /srv/redmine-3.1# RAILS_ENV="production" rake --trace redmine:check_periodictasks 
** Invoke redmine:check_periodictasks (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute redmine:check_periodictasks
````

**Comment by jperelli** [comment]
Thank you for the confirmation, I'm merging it
